gamemodes/terrortown/gamemode/client/cl_voice.lua

```diff
@@ -350,75 +289,33 @@ local function ReceiveVoiceState()
     end
 
     ply[tm .. "_gvoice"] = isGlobal
-
-    if not IsValid(PlayerVoicePanels[ply]) then
-        return
-    end
-
-    PlayerVoicePanels[ply].Color = isGlobal and VP_GREEN or (ply:GetRoleColor() or VP_RED)
 end
 net.Receive("TTT_RoleVoiceState", ReceiveVoiceState)
 
-local function VoiceClean()
-    if not PlayerVoicePanels then
-        return
-    end
-
-    for ply, pnl in pairs(PlayerVoicePanels) do
-        if IsValid(pnl) and IsValid(ply) then
-            continue
-        end
-
-        GAMEMODE:PlayerEndVoice(ply)
-    end
-end
-timer.Create("VoiceClean", 10, 0, VoiceClean)
-
 ---
 -- Called when @{Player} stops using voice chat.
 -- @param Player ply @{Player} who stopped talking
--- @param boolean no_reset whether the stored voice state shouldn't reset
 -- @hook
 -- @realm client
 -- @ref https://wiki.facepunch.com/gmod/GM:PlayerEndVoice
 -- @local
-function GM:PlayerEndVoice(ply, no_reset)
-    if IsValid(PlayerVoicePanels[ply]) then
-        PlayerVoicePanels[ply]:Remove()
-
-        PlayerVoicePanels[ply] = nil
+function GM:PlayerEndVoice(ply)
+    if not IsValid(ply) then
+        return
     end
 
-    if IsValid(ply) and not no_reset then
-        local tm = ply:GetTeam()
+    local plyTeam = ply:GetTeam()
 
-        if tm ~= TEAM_NONE and not TEAMS[tm].alone then
-            ply[tm .. "_gvoice"] = false
-        end
+    -- if someone else than us stops talking, the UI should be updated as well
+    if ply ~= LocalPlayer() then
+        ply:SetSpeakingInVoice(false)
     end
 
-    if ply == LocalPlayer() then
-        VOICE.SetSpeaking(false)
+    if plyTeam ~= TEAM_NONE and not TEAMS[plyTeam].alone then
+        ply[plyTeam .. "_gvoice"] = false
```

Comment:
Why are we using this weird variable instead of the speaking attribute here?

Reply:
I've struggled a bit to understand this as well, but chose to keep it as is and only change the UI. So this weird code stayed as is without a change.

The git-diff is a bit misleading here as I did not change anything related to the core chat logic